.github/workflows/apps-cypress-percy-test.yml

```diff
@@ -10,45 +10,65 @@ on:
       - "apps/**"
     branches: [main]
 
+permissions:
+  contents: read
+
+concurrency:
+  group: site-e2e-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
 jobs:
   cypress-run:
     if: github.event.pull_request.draft == false
     runs-on: ubuntu-latest
     strategy:
+      fail-fast: false
       matrix:
-        app: [ff-site, ffdweb-site, filecoin-site]
+        include:
+          - app: ff-site
+            ready_path: /blog
+            percy_token: PERCY_TOKEN_FF_SITE
+          - app: ffdweb-site
+            ready_path: /blog
+            percy_token: PERCY_TOKEN_FFDWEB_SITE
+          - app: filecoin-site
+            ready_path: /blog
+            percy_token: PERCY_TOKEN_FILECOIN_SITE
     defaults:
       run:
         working-directory: apps/${{ matrix.app }}
 
     env:
       AIRTABLE_READ_ONLY_TOKEN: ${{ secrets.AIRTABLE_READ_ONLY_TOKEN }}
-      PERCY_TOKEN: ${{ matrix.app == 'ff-site' && secrets.PERCY_TOKEN_FF_SITE || secrets.PERCY_TOKEN_FFDWEB_SITE || '' }}
+      BASE_URL: "http://localhost:3000"
       ENCRYPTION_SECRET_KEY: "a-random-key-so-that-ci-builds-dont-fail"
+      PERCY_TOKEN: ${{ secrets[matrix.percy_token] }}
```

Comment:
Duplicate `PERCY_TOKEN definition`: `PERCY_TOKEN` at `env`: and again in the Percy step. Keep just the step-local one (least exposure in logs/env dumps).

```suggestion
```